goth/runner/log.py

```diff
@@ -148,16 +155,25 @@ def configure_logging_for_test(test_log_dir: Path) -> None:
             api_monitor_logger.handlers.remove(proxy_handler)
 
 
-class MonitorHandler(logging.Handler):
-    """A logging handler that passes messages from log records to an event monitor."""
+class MonitoringFilter(logging.Filter):
```

Comment:
Here we need an object that will
* perform some extra processing for a given log record (i.e. add its message to a monitor)
* modify the record (i.e. add colors) for upstream loggers

A `logging.Handler` can only do the first of those two things.